### PR TITLE
XRAYJENKINS-108

### PR DESCRIPTION
**Pull request by fabiotcunha28**

### CODE REVIEW

File: src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Jenkins.class, ProxyConfiguration.class})
+@PowerMockIgnore({"javax.net.ssl.*", "javax.management.*"})
+public class ProxyUtilTest {
+
+    public static final String PROXY_HOSTNAME = "whatever";
```
Comment (Russell616) [inline src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java:29]: All these variables could be private

**Review by Russell616 — CHANGES_REQUESTED** [review]

File: src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java (hunk 1 above)
Comment (fabiotcunha28) [reply src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java]: Done.

File: src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    private XrayClientImpl xrayClient;
+
+    @Before
+    public void setup() throws IOException {
+        PowerMockito.mockStatic(Jenkins.class);
+        PowerMockito.mockStatic(ProxyConfiguration.class);
+
+        Whitebox.setInternalState(jenkins, "proxy", proxyConfiguration);
+        Whitebox.setInternalState(jenkins, "proxy", proxyConfiguration);
```
Comment (Russell616) [inline src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java:51]: this line is cloned, can this line be removed?

**Review by Russell616 — CHANGES_REQUESTED** [review]

File: src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java (hunk 2 above)
Comment (fabiotcunha28) [reply src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java]: Yes, it was added by accident.

File: src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+@PrepareForTest({Jenkins.class, ProxyConfiguration.class})
+@PowerMockIgnore({"javax.net.ssl.*", "javax.management.*"})
+public class ProxyUtilTest {
+
+    private static final String PROXY_DOMAIN = "proxydomain.com";
+    private static final int PROXY_PORT = 8080;
+    private static final String JIRA_DOMAIN = "jiradomain.com";
+    private static final int JIRA_PORT = 8080;
+    private static final String JIRA_HTTP_URL = "http://" + JIRA_DOMAIN;
+    private static final String JIRA_HTTP_URL_WITH_PORT = "http://" + JIRA_DOMAIN + ":" + JIRA_PORT;
```
Comment (Russell616) [inline src/test/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtilTest.java:33]: Please add a "legacy check" to the unit tests:
Before this update, the "No Proxy Host" was in the following format: *www.my-jira-server.com/*
Please check this old format still works and is still being ignored

**Review by Russell616 — CHANGES_REQUESTED** [review]

**Review by Russell616 — APPROVED** [review]

**Pull request merged**